### ISSUE TITLE
Minor coding style issues

### ISSUE THREAD

**Issue by Exirel** [MagicTrucy/cryptolib#2]
I was looking at the tests to understand your issue with PYTHONPATH, and I found these minor coding style issues (mostly coding style but not only):

### .strip('\n')

Source: https://github.com/MagicTrucy/cryptolib/blob/013e2a72734074b25c96549cabbc9be721a94a05/cryptolib/cryptotest.py#L70

I'm not sure what is your intent here. Just to be sure, here is a short comparison between `str.strip()` and `str.strip('\n')`:

```
>>> 'toto\n'.strip('\n')
'toto'
>>> 'toto\n   '.strip('\n')
'toto\n   '
>>> 'toto   \n'.strip('\n')
'toto   '
>>> 'toto\n'.strip()
'toto'
>>> 'toto\n   '.strip()
'toto'
>>> 'toto   \n'.strip()
'toto'
```

I'm not sure you need the `\n` as argument. Maybe a simple `.strip()` is far enough.

### File descriptor opened but never closed

Source: https://github.com/MagicTrucy/cryptolib/blob/013e2a72734074b25c96549cabbc9be721a94a05/cryptolib/cryptotest.py#L67

You should not (some say must not) open a file descriptor without closing it explicitly. This is considered bad practice to let the process to close the file descriptor when shut down.

I suggest to use the widely popular way of preventing such issue:

```
with open(filename) as file_descriptor:
    # your code here
```

Also, it's common to call that `file_descriptor` variable something like `fd`, or `file_descriptor`, and not `file` only - but that's more a matter of taste, I guess. Simply know that `file` is less explicit than `file_descriptor`, as a `file` may be the filename, the file path, or the content itself - it's too ambiguous.

### Use more list-comprehension

Source: https://github.com/MagicTrucy/cryptolib/blob/013e2a72734074b25c96549cabbc9be721a94a05/cryptolib/cryptotest.py#L67

In this case, your for-loop - while totally readable - is a very good candidate for a list-comprehension. It's always a good idea to get familiarized with such pythonic coding style, as it's considered more elegant. As a matter of fact, this is also more efficient, since it doesn't require to call N times the `list.append()` method:

```
candidates = [
    cryptolib.hex_to_bytes(line.strip('\n'))
    for line in file
]
```

Don't worry if you are not familiar with list-comprehensions. It's OK, it took me a while before I could really understand them. It's a good practice to know them and to use them in the appropriate context. I strongly advise you to dive into them whenever you can!

### String concatenation

Source: https://github.com/MagicTrucy/cryptolib/blob/013e2a72734074b25c96549cabbc9be721a94a05/cryptolib/cryptotest.py#L87

You do not need the + sign whenever you want to concatenate two string declaration:

```
>>> 'hello' 'world'
'helloworld'
>>> 'hello' + 'world'
'helloworld'
>>> ('hello'
 'world')
'helloworld'
>>> ('hello'
 + 'world')
'helloworld'
```

It's considered good practice not to use the + sign in these cases. No visual noises!

### Do not hide python string encoding logic

Source: https://github.com/MagicTrucy/cryptolib/blob/013e2a72734074b25c96549cabbc9be721a94a05/cryptolib/cryptolib.py#L29

I consider bad practice to hide encoding str to bytes and decoding  bytes to str inside custom function. This put an unnecessary step between how python process str/bytes objects and how your program will use them.

`str.encode` and `bytes.decode` should be always as explicit as possible, in a way that let the reader understand that:

1. the object called is always a string or a bytes
2. the charset used is always explicit and not hidden

I prefer this:

```
data.decode('utf-8')
```

Rather than:

```
bytes_to_string(data)
```

Note that your function forbid to use non-bytes/str object types, while using the encoding/decoding methods allow to use any object that follow the right interface.

In python, we prefer follow interface than type.

TL;DR: helper methods are confusing because they hide basic python logic that might be required for developers to understand what's going on, and they may prevent usage that are very common in the pythonic world.

---

Nothing here is important or urgent. It's just my point of view as an experienced python programmer.
Also, very nice coding style, it's clean and very readable. Not all beginners can do that!

Good luck, have fun!

**Comment by MagicTrucy** [comment]
Right, so most of the problems come from the fact that:
- I didn't pylint'd cryptotest.py
- My IDE (IntelliJ IDEA with the Python plugin) only highlights PEP 8 for Python 2 errors

/me locks LOIC on IntelliJ IDEA

**Comment by Exirel** [comment]
Ah ah, same here, I always forgot!

**Comment by MagicTrucy** [comment]
There! Took care of most of the things you pointed out!
Except for the python string encoding logic because I like having clear function names (and I already have to deal with other aspects of the library).

For the record, if you install the Python 3 versions of pylint and pep8, IntelliJ IDEA will use them automagically.
